securepass/djangoauth.py

```diff
@@ -1,44 +1,46 @@
-##
-## Django SecurePass APIs authentication backend
-##
-## This code is released under GPLv2
-##
-## (c) 2013 Giuseppe Paterno' (gpaterno@gpaterno.com)
-##          GARL Sagl (www.garl.ch)
-##
-
-from securepass import securepass
+# Django SecurePass APIs authentication backend
+#
+# This code is released under GPLv2
+#
+# (c) 2013 Giuseppe Paterno' (gpaterno@gpaterno.com)
+#          GARL Sagl (www.garl.ch)
+#
+
+from . import securepass
 from django.conf import settings
 from django.contrib.auth.models import User
 
+
 class SecurePassAuthBackend(object):
 
     """ SecurePass' core authentication backend
     Settings:
-        SP_APP_ID -> SecurePass APP ID
-        SP_APP_SECRET -> SecurePass APP Secret
-        SP_ENDPOINT -> API endpoint
-        SP_AUTOCREATE_USER -> True/False, set if you want automatic creation of user
+        SP_APP_ID          -> SecurePass APP ID
+        SP_APP_SECRET      -> SecurePass APP Secret
+        SP_ENDPOINT        -> API endpoint
+        SP_AUTOCREATE_USER -> True/False, set if you want automatic creation
+                              of user
+        SP_REALM           -> SecurePass realm
     """
 
-    ## Actual auth
+    # Actual auth
     def authenticate(self, username, password):
 
-        try:
-            endpoint = settings.SP_ENDPOINT
-            autocreate = settings.SP_AUTOCREATE_USER
-        except:
-            endpoint = "https://beta.secure-pass.net/"
-            autocreate = True
-
-        sp_handle = securepass.SecurePass(app_id=settings.SP_APP_ID,
-                                          app_secret=settings.SP_APP_SECRET,
-                                          endpoint=endpoint)
+        endpoint = getattr(
+            settings, 'SP_ENDPOINT', 'https://beta.secure-pass.net')
+        autocreate = getattr(settings, 'SP_AUTOCREATE_USER', True)
 
+        sp_handle = securepass.SecurePass(
+            app_id=getattr(settings, 'SP_APP_ID', ''),
```

Comment:
For the moment let's not change the behaviour too much, but in a next patch it would be nice to raise an ```ImproperlyConfigured Exception```, as these two parameters are mandatory to connect to the service.